_posts/2015-07-13-type-members-parameters.md

```diff
@@ -0,0 +1,227 @@
+---
+layout: post
+title: Type members are [almost] type parameters
+
+meta:
+  nav: blog
+  author: S11001001
+  pygments: true
+---
+
+Type members are [almost] type parameters
+=========================================
+
+*This is the first of a series of articles on “Type Parameters and Type
+Members”.*
+
+Type members like `Member`
+
+```scala
+class Blah {
+  type Member
+}
+```
+
+and parameters like `Param`
+
+```scala
+class Blah2[Param]
+```
+
+have more similarities than differences.  The choice of which to use
+for a given situation is usually a matter of convenience.  In brief, a
+rule of thumb: **if you intend to use a parameter existentially in
+most cases, changing it to a member is probably better; a parameter is
+more convenient and harder to screw up in most circumstances**.
```

Comment:
On second thought it might be better to reverse this statement:

*In most cases a type parameter is more convenient and harder to screw up. However, if you intend to primarily leave a type existential, a type member is probably better.*